ads/google/csa.js

```diff
@@ -0,0 +1,327 @@
+      let newHeight = container.style.height;
+
+      // In older versions of iOS, this height will be different because the
+      // container height is resized.
+      // In Chrome and iOS 10.0.2 the height is the same because
+      // the container isn't resized.
+      if (oldHeight != newHeight && newHeight != currentAmpHeight) {
+
+        // style.height returns "60px" (for example), so turn this into an int
+        newHeight = parseInt(newHeight, 10);
+
+        // Also update the onclick function to resize to the right height.
+        const overflow = global.document.getElementById('overflow');
+        if (overflow) {
+          overflow.onclick = function() {
+            global.context.requestResize(undefined, newHeight);
+          };
+        }
+
+        // Resize the container to the correct height
+        global.context.requestResize(undefined, newHeight);
+      }
+
+    }, orientationChangeTimeout);
+
+  }, false);
+
+  // Only call for ads once the script has loaded
+  loadScript(global, 'https://www.google.com/adsense/search/ads.js', () => {
+
+    // Make the call for CSA ads
+    // Call the right product based on arguments passed
+    if (data['afsPageOptions'] != null && data['afshPageOptions'] == null) {
+      // AFS only
+
+      // Add a callback without any backfill options
+      afsAdblockOptions['adLoadedCallback'] =
+        resizeIframe.bind(null, global, null, null);
+
+      global._googCsa('ads', afsPageOptions, afsAdblockOptions);
+
+    } else if (data['afsPageOptions'] == null
+      && data['afshPageOptions'] != null) {
+      // AFSH only
+
+      // Create the callback without any backfill options
+      afshAdblockOptions['adLoadedCallback'] =
+        resizeIframe.bind(null, global, null, null);
+
+      global._googCsa('plas', afshPageOptions, afshAdblockOptions);
+
+    } else if (data['afsPageOptions'] != null
+      && data['afshPageOptions'] != null) {
+      // AFSh backfilled with AFS
+
+      // Create a callback with the AFS options
+      afshAdblockOptions['adLoadedCallback'] =
+        resizeIframe.bind(null, global, afsPageOptions, afsAdblockOptions);
+
+      global._googCsa('plas', afshPageOptions, afshAdblockOptions);
+    }
+  });
+
+}
+
+/**
+ * CSA callback function to resize the iframe and/or request backfill
+ * @param {?Object} backfillPageOptions AFS page options (if necessary)
+ * @param {?Object} backfillAdblockOptions AFS ad unit options (if necessary)
+ * @param {string} containerName
+ * @param {boolean} adsLoaded
+ * @visibleForTesting
+ */
+export function resizeIframe(global, backfillPageOptions,
+  backfillAdblockOptions, containerName, adsLoaded) {
+
+  if (adsLoaded) {
+
+    // Get actual height and width of container
+    const container = global.document.getElementById(containerName);
+    const height = container.offsetHeight;
+    currentAmpHeight =
+        global.context.initialIntersection.boundingClientRect.height;
```

Comment:
Actually, global.innerHeight doesn't work because I need the height of the AMP container at the time of requesting the ads before the iframe is resized.  Different browsers have different behaviors (some bust the iframe, some don't).  Changing to global.innerHeight broke the overflow implementation on iOS.